src/main/java/junmin/netflix/costreportservice/controller/CostReportController.java

```diff
@@ -0,0 +1,78 @@
+package junmin.netflix.costreportservice.controller;
+
+import io.swagger.annotations.ApiOperation;
+import junmin.netflix.costreportservice.exception.InvalidProductionException;
+import junmin.netflix.costreportservice.exception.InvalidReportException;
+import junmin.netflix.costreportservice.pojo.EPCostRecord;
+import junmin.netflix.costreportservice.service.EPCostReportService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+public class CostReportController {
+	private EPCostReportService service;
+	private final Logger LOGGER = LoggerFactory.getLogger(CostReportController.class);
+
+	public CostReportController(EPCostReportService service){
+		this.service = service;
+	}
+
+	@ApiOperation(value="process single EP cost report", notes="process single EP cost report")
+	@PostMapping(path="/report/production/{prodName}/ep",consumes = "application/json", produces = "application/json")
+	public ResponseEntity<List<EPCostRecord>> postEPCostReport(@PathVariable("prodName") String prodName, @RequestBody List<EPCostRecord> costs){
+		try {
+			List<EPCostRecord> result = service.processEPCostReport(prodName, costs);
+			return ResponseEntity.ok(result);
+		}catch(InvalidReportException e){
+			LOGGER.error("POST /api/report/production/"+prodName+"/ep failed: " + e.getMessage());
```

Comment:
yes, we can have parameterized logging here using {} as a placeholder:
LOGGER.error("/api/report/{}/ep failed", prodName)

And as Paul pointed out in another place, I can use HttpServletRequest itself to get the path, instead of hard code here. 

As to the benefit of using parameterized logging is for better readability, neater code, and avoid string1+string2+string3..., String is immutable, this type of concatenation should be avoided.